File: server/src/models/Artist.ts
import {
    Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany,
} from 'typeorm';
import Genre from './Genre';
import Album from './Album';

@Entity()
class Artist {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    name: string;

    @Column()
    imageUrl: string;

    @ManyToOne((type) => Genre, (genre) => genre.artists)
    genre: Genre;

    @OneToMany((type) => Album, (album) => album.genre)
    albums: Album[];
}

export default Artist;

File: MiniVibe/MiniVibe/Models/Artist.swift
//
//  Artist.swift
//  MiniVibe
//
//  Created by TTOzzi on 2020/12/09.
//

import Foundation

struct Artists: Decodable {
    let data: [Artist]
}

struct Artist: Decodable {
    let id: Int
    let name: String
    let imageUrl: String
    let genre: Genre
}

struct ArtistInfo: Decodable {
    struct Album: Decodable {
        let id: Int
        let title: String
        let imageUrl: String
    }
    
    let id: Int
    let name: String
    let imageUrl: String
    let genre: Genre
    let tracks: [TrackInfo]
    let albums: [Album]
}

struct Genre: Decodable {
    let name: String
}
